### PR TITLE
[monorepo] change getFreeBalance return type

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
This PR changes the return type of the `GetFreeBalance` node RPC method from

```
{
  alice: string;
  bob: string;
  aliceBalance: BigNumber;
  bobBalance: BigNumber;
}
``` 

to

```
{
    [s: string]: BigNumber;
};
```

The motivation is to a more stable return type so that the ETHBucket contract can be upgraded (e.g. Interpreters, >2-party channels) independently of the consumers of the `GetFreeBalance` RPC method.

### CODE REVIEW

File: packages/high-roller-bot/src/bot.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ function respond(node: Node, nodeAddress: Address, msg: UpdateStateMessage) {
     console.log(newState);
   }
 
-  if (newState.stage === 2) {
+  if (newState["stage"] === 2) {
```
Comment (snario) [inline packages/high-roller-bot/src/bot.ts]: I think you could introduce a type for this inside the scope of the `high-roller-bot` package and type `newState` as that; `HighRollerAppState`.

File: packages/high-roller-bot/src/utils.ts [hunk 2]
```diff
@@ -49,22 +50,29 @@ export async function fetchMultisig(baseURL: string, token: string) {
...
   amount: string,
-  multisigAddress: string,
-  skip: boolean = false
+  multisigAddress: string
 ) {
-  if (skip) {
-    return;
+  const myFreeBalanceAddress = fromExtendedKey(
+    node.publicIdentifier
+  ).derivePath("0").address;
```
Comment (snario) [inline packages/high-roller-bot/src/utils.ts]: I think the `node` object has a method to retrieve this, no? If not, we should add it.

File: packages/high-roller-bot/src/utils.ts [hunk 3]
```diff
@@ -49,22 +50,29 @@ export async function fetchMultisig(baseURL: string, token: string) {
...
+
+  const preDepositBalances = await getFreeBalance(node, multisigAddress);
+
+  if (Object.keys(preDepositBalances).length !== 2) {
+    throw Error("Unexpected number of entries");
```
Comment (snario) [inline packages/high-roller-bot/src/utils.ts]:
```suggestion
    throw new Error("Unexpected number of entries");
```

File: packages/high-roller-bot/src/utils.ts [hunk 4]
```diff
@@ -49,22 +50,29 @@ export async function fetchMultisig(baseURL: string, token: string) {
...
+  }
+
+  if (!preDepositBalances[myFreeBalanceAddress]) {
+    throw Error("My address not found");
```
Comment (snario) [inline packages/high-roller-bot/src/utils.ts]:
```suggestion
    throw new Error("My address not found");
```

File: packages/high-roller-bot/src/utils.ts [hunk 5]
```diff
@@ -49,22 +50,29 @@ export async function fetchMultisig(baseURL: string, token: string) {
...
   }
 
-  const { aliceBalance, bobBalance } = (await getFreeBalance(
-    node,
-    multisigAddress
-  )).state;
+  const [counterpartyFreeBalanceAddress] = Object.keys(
```
Comment (snario) [inline packages/high-roller-bot/src/utils.ts]:
```suggestion
  const [ counterpartyFreeBalanceAddress ] = Object.keys(
```

File: packages/high-roller-bot/src/utils.ts [hunk 6]
```diff
@@ -78,42 +86,30 @@ export async function deposit(
...
-        Updated Free Balance: alice: ${updatedAliceBalance}, bob: ${updatedBobBalance}
-      `);
+    const postDepositBalances = await getFreeBalance(node, multisigAddress);
+
+    if (
+      !postDepositBalances[myFreeBalanceAddress].gt(
+        preDepositBalances[myFreeBalanceAddress]
+      )
+    ) {
+      throw Error("My balance was not updated.");
```
Comment (snario) [inline packages/high-roller-bot/src/utils.ts]:
```suggestion
      throw Error("My balance was not increased.");
```

File: packages/high-roller-bot/src/utils.ts [hunk 7]
```diff
@@ -78,42 +86,30 @@ export async function deposit(
...
+    }
+
+    console.info("Waiting for counter party to deposit same amount");
+
+    while (
+      !(await getFreeBalance(node, multisigAddress))[
+        counterpartyFreeBalanceAddress
+      ].gt(preDepositBalances[counterpartyFreeBalanceAddress])
+    ) {
```
Comment (snario) [inline packages/high-roller-bot/src/utils.ts]: Wrap this 3 line conditional in a named function e.g., `checkFreeBalanceGreaterThan(...)`

File: packages/node/src/methods/app-instance/get-free-balance/controller.ts [hunk 8]
```diff
@@ -25,9 +25,16 @@ export default class GetFreeBalanceController extends NodeController {
 
     const stateChannel = await store.getStateChannel(multisigAddress);
 
+    const {
+      alice,
+      bob,
+      aliceBalance,
+      bobBalance
+    } = stateChannel.getFreeBalanceFor(AssetType.ETH)
+      .state as ETHBucketAppState;
```
Comment (snario) [inline packages/node/src/methods/app-instance/get-free-balance/controller.ts]: Split up lines; looks awkward

File: packages/tic-tac-toe-bot/src/utils.ts [hunk 9]
```diff
@@ -78,42 +86,30 @@ export async function deposit(
...
+      throw Error("My balance was not updated.");
+    }
+
+    console.info("Waiting for counter party to deposit same amount");
+
+    while (
+      !(await getFreeBalance(node, multisigAddress))[
+        counterpartyFreeBalanceAddress
+      ].gt(preDepositBalances[counterpartyFreeBalanceAddress])
+    ) {
```
Comment (snario) [inline packages/tic-tac-toe-bot/src/utils.ts]: Same as earlier

File: packages/types/src/node-protocol.ts [hunk 10]
```diff
@@ -100,7 +99,7 @@ export namespace Node {
   };
 
   export type GetFreeBalanceStateResult = {
-    state: ETHBucketAppState;
+    [ s:string ]: BigNumber
```
Comment (snario) [inline packages/types/src/node-protocol.ts]:
```suggestion
    [ s: string ]: BigNumber
```

**Comment by snario** [comment]
Can you describe the motivation for this change for others to read / review

File: packages/high-roller-bot/src/utils.ts (hunk 5 above)
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/high-roller-bot/src/utils.ts]: This change is reverted by the linter

File: packages/node/src/methods/app-instance/get-free-balance/controller.ts (hunk 8 above)
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/node/src/methods/app-instance/get-free-balance/controller.ts]: this is what happens when I lint-fix

```
    const { alice, bob, aliceBalance, bobBalance } = stateChannel.getFreeBalanceFor(AssetType.ETH).state as ETHBucketAppState;
```

**Pull request merged**